webapps/frontend/ui/cockpit/client/scripts/directives/time-to-live/time-to-live-dialog.js

```diff
@@ -42,8 +42,42 @@ const Controller = [
     $scope.ttl = $scope.definition.historyTimeToLive;
 
     $scope.openBatchOperation = () => {
+      let batchSearchQuery = null;
+      const operation =
+        $scope.$parent?.resource === 'process-definition'
+          ? 'PROCESS_SET_REMOVAL_TIME'
+          : 'DECISION_SET_REMOVAL_TIME';
+
+      if ($scope.$parent?.resource === 'process-definition') {
+        batchSearchQuery = JSON.stringify([
+          {
+            type: 'PIprocessDefinitionKey',
+            operator: 'eq',
+            value: $scope.definition.key
+          },
+          {
+            type: 'PIfinished',
+            operator: 'eq',
+            value: ''
+          }
+        ]);
+      }
+
+      if ($scope.$parent?.resource === 'decision-definition') {
```

Comment:
🔧 If there are no other cases, it could be simplified.

```suggestion
      } else {
```